lib/ExpensiMark.js

```diff
@@ -74,7 +74,12 @@ export default class ExpensiMark {
 
             {
                 name: 'heading1',
-                regex: /^# +(?! )((?:(?!<pre>|\n|\r\n).)+)/gm,
+                process: (textToProcess, replacement, shouldKeepWhitespace = false) => {
+                    if (shouldKeepWhitespace) {
+                        return textToProcess.replace(/^# ( *(?! )(?:(?!<pre>|\n|\r\n).)+)/gm, replacement);
+                    }
+                    return textToProcess.replace(/^# +(?! )((?:(?!<pre>|\n|\r\n).)+)/gm, replacement);
```

Comment:
Instead of calling `.replace` in two places, can we assign `regex` conditionally, i.e.
```suggestion
                    const regexp = shouldKeepWhitespace
                        ? /^# ( *(?! )(?:(?!<pre>|\n|\r\n).)+)/gm
                        : /^# +(?! )((?:(?!<pre>|\n|\r\n).)+)/gm;
                    return textToProcess.replace(regexp, replacement);
```